modules/pam_access/pam_access.c

```diff
@@ -154,6 +159,73 @@ parse_args(pam_handle_t *pamh, struct login_info *loginfo,
     return 1;  /* OK */
 }
 
+/* --- evaluting all files in VENDORDIR/security/access.d and /etc/security/access.d --- */
+static int
+compare_filename(const void *a, const void *b)
+{
+	return strcmp(basename(* (char * const *) a), basename(* (char * const *) b));
```

Comment:
This function shall not modify data, so please don't use the XPG version of `basename`.
If you cannot use the GNU version of `basename`, don't use basename at all.